trio/tests/test_socket.py

```diff
@@ -990,3 +990,26 @@ async def receiver():
             nursery.start_soon(receiver)
             await wait_all_tasks_blocked()
             a.close()
+
+
+async def test_many_sockets():
+    total = 5000  # Must be more than MAX_AFD_GROUP_SIZE
+    sockets = []
+    for x in range(total // 2):
+        try:
+            a, b = stdlib_socket.socketpair()
+        except OSError as e:
+            if e.errno != 24:
```

Comment:
Is this an error you hit on Linux? It looks like on Linux this is `errno.EMFILE`, which would make sense, since `EMFILE` means "you hit the limit on the number of file descriptors that a single process is allowed to have open". (There's also the confusingly-similar `ENFILE`, which means "you hit some kind of systemwide limit on the number of file descriptors".) So at the least, we should change this to `errno.EMFILE` instead of having a magic integer in the source code :-). And probably add `ENFILE` as an option too. Or maybe simplify to

```
except OSError as exc:
    assert exc.errno in (errno.EMFILE, errno.ENFILE)
    break
```

BTW, a handy trick: if you have some errno number and you want to figure out what the error code is, you can do:

```python
>>> import errno
>>> errno.errorcode[24]
'EMFILE'
```